vmdb/app/controllers/application_controller.rb

```diff
@@ -2168,6 +2168,10 @@ def prov_redirect(typ=nil)
         render_flash_not_applicable_to_model(typ)
         return
       end
+      if typ.eql?("migrate") && VmOrTemplate.where(:id => vm_ids, :type => %w(VmMicrosoft)).exists?
```

Comment:
Also, no need for the Array on the type, in fact you should not query on the type at all.

```ruby
if typ == "migrate" && VmMicrosoft.where(:id => vm_ids).exists?
```

However, moving forward we should stop hardcoding provider specific checks into the UI, because it makes pluggability so much harder to untangle.  This particular case could easily be handled with a `supports_migrate?` method in the model, and then (given you have the vm objects) the UI could would be

```ruby
if typ == "migrate" && vms.any?(&:supports_migrate?)
```

If you don't have the objects, but just the ids, then you can do something like

```ruby
models_supporting_migrate = VmOrTemplate.subclasses.select(&:supports_migrate?)
if typ == "migrate" && VmOrTemplate.where(:id => vm_ids, :type => models_supporting_migrate).exists?
```

cc @blomquisg @chessbyte